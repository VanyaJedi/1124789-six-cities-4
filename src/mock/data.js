export const offers = [
  {
    id: `1`,
    img: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-02.jpg`],
    title: `Beautiful & luxurious apartment at great location`,
    desc: `Beautiful & luxurious apartment at great location, Beautiful & luxurious apartment at great location`,
    isPrime: true,
    type: `room`,
    rate: 3.1,
    bedAmount: 2,
    maxAdults: 3,
    cost: `100`,
    houseItems: [`TV`, `Rope`],
    owner: {
      avatar: `img/avatar-angelina.jpg`,
      name: `Ivan`,
      isSuper: false
    },
    coordinates: [52.3909553943508, 4.85309666406198]
  },
  {
    id: `2`,
    img: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-02.jpg`],
    title: `Beautiful & luxurious apartment at great location`,
    desc: `Beautiful & luxurious apartment at great location, Beautiful & luxurious apartment at great location`,
    isPrime: true,
    type: `room`,
    rate: 3.1,
    bedAmount: 2,
    maxAdults: 3,
    cost: `100`,
    houseItems: [`TV`, `Rope`],
    owner: {
      avatar: `img/avatar-angelina.jpg`,
      name: `Ivan`,
      isSuper: false
    },
    coordinates: [52.369553943508, 4.85309666406198]
  },
  {
    id: `3`,
    img: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-02.jpg`],
    title: `Beautiful & luxurious apartment at great location`,
    desc: `Beautiful & luxurious apartment at great location, Beautiful & luxurious apartment at great location`,
    isPrime: true,
    type: `room`,
    rate: 3.3,
    bedAmount: 2,
    maxAdults: 3,
    cost: `100`,
    houseItems: [`TV`, `Rope`],
    owner: {
      avatar: `img/avatar-angelina.jpg`,
      name: `Ivan`,
      isSuper: false
    },
    coordinates: [52.3909553943508, 4.929309666406198]
  },
  {
    id: `4`,
    img: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-02.jpg`],
    title: `Beautiful & luxurious apartment at great location`,
    desc: `Beautiful & luxurious apartment at great location, Beautiful & luxurious apartment at great location`,
    isPrime: true,
    type: `room`,
    rate: 3.1,
    bedAmount: 2,
    maxAdults: 3,
    cost: `100`,
    houseItems: [`TV`, `Rope`],
    owner: {
      avatar: `img/avatar-angelina.jpg`,
      name: `Ivan`,
      isSuper: false
    },
    coordinates: [52.3809553943508, 4.939309666406198]
  },
];


export const reviews = [
  {
    id: `1`,
    text: `bla bla bla bla bla bla bla blablabla`,
    rate: 5,
    userName: `Olya`,
    date: new Date(`1995-12-17`)
  },
  {
    id: `2`,
    text: `olololololololo`,
    rate: 3,
    userName: `Masha`,
    date: new Date(`1995-12-17`)
  },
  {
    id: `3`,
    text: `bla bla bla bla bla bla bla blablabla`,
    rate: 5,
    userName: `Olya`,
    date: new Date(`1995-12-17`)
  },
  {
    id: `4`,
    text: `olololololololo`,
    rate: 3,
    userName: `Masha`,
    date: new Date(`1995-12-17`)
  },
  {
    id: `5`,
    text: `bla bla bla bla bla bla bla blablabla`,
    rate: 5,
    userName: `Olya`,
    date: new Date(`1995-12-17`)
  },
  {
    id: `6`,
    text: `olololololololo`,
    rate: 3,
    userName: `Masha`,
    date: new Date(`1995-12-17`)
  },
  {
    id: `7`,
    text: `bla bla bla bla bla bla bla blablabla`,
    rate: 5,
    userName: `Olya`,
    date: new Date(`1995-12-17`)
  },
  {
    id: `8`,
    text: `olololololololo`,
    rate: 3,
    userName: `Masha`,
    date: new Date(`1995-12-17`)
  },
  {
    id: `9`,
    text: `bla bla bla bla bla bla bla blablabla`,
    rate: 5,
    userName: `Olya`,
    date: new Date(`1995-12-17`)
  },
  {
    id: `10`,
    text: `olololololololo`,
    rate: 3,
    userName: `Masha`,
    date: new Date(`1995-12-17`)
  },
  {
    id: `11`,
    text: `bla bla bla bla bla bla bla blablabla`,
    rate: 5,
    userName: `Olya`,
    date: new Date(`1995-12-17`)
  },
  {
    id: `12`,
    text: `olololololololo`,
    rate: 3,
    userName: `Masha`,
    date: new Date(`1995-12-17`)
  },
  {
    id: `13`,
    text: `bla bla bla bla bla bla bla blablabla`,
    rate: 5,
    userName: `Olya`,
    date: new Date(`1995-12-17`)
  },
  {
    id: `14`,
    text: `olololololololo`,
    rate: 3,
    userName: `Masha`,
    date: new Date(`2005-11-16`)
  },
];
