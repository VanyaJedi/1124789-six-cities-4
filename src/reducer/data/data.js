
import {extend} from "../../utils.js";

const initialState = {
  city: null,
  offers: [],
  offersOnCity: [],
  reviews: [],
};

const actionType = {
  INIT_CITY: `INIT_CITY`,
  CITY_CHANGE: `CITY_CHANGE`,
  LOAD_REVIEWS: `LOAD_REVIEWS`,
  LOAD_OFFERS: `LOAD_OFFERS`,
  CITY_OFFERS: `CITY_OFFERS`
};

const actionCreator = {

  changeCity: (city) => ({
    type: actionType.CITY_CHANGE,
    payload: city
  }),

  loadReviews: (reviews) => {
    return {
      type: actionType.LOAD_REVIEWS,
      payload: reviews,
    };
  },

  loadOffers: (offers) => {
    return {
      type: actionType.LOAD_OFFERS,
      payload: offers,
    };
  },

  cityOffers: (offers) => {
    return {
      type: actionType.CITY_OFFERS,
      payload: offers,
    };
  }
};


const Operation = {
  loadOffers: () => (dispatch, getState, api) => {
    return api.get(`/hotels`)
      .then((response) => {
        dispatch(actionCreator.changeCity(response.data[0].city));
        dispatch(actionCreator.loadOffers(response.data));
      });
  },
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case actionType.INIT_CITY:
      return extend(state, {city: action.payload});
    case actionType.CITY_CHANGE:
      return extend(state, {city: action.payload});
    case actionType.LOAD_REVIEWS:
      return extend(state, {reviews: action.payload});
    case actionType.LOAD_OFFERS:
      return extend(state, {offers: action.payload});
    case actionType.CITY_OFFERS:
      return extend(state, {offersOnCity: action.payload});
    default:
      return state;
  }
};

export {reducer, Operation, actionType, actionCreator};
