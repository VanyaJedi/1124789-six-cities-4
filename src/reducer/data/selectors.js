
import {createSelector} from "reselect";
import nameSpace from "../name-space.js";
import {createOffer, createCity} from "../../adapters/offers.js";

const CITIES_TO_SHOW = 6;
const NAME_SPACE = nameSpace.DATA;

export const getCity = (state) => {
  return createCity(state[NAME_SPACE].city);
};

export const getOffers = (state) => {
  return state[NAME_SPACE].offers.map((offer) => createOffer(offer));
};

export const getReviews = (state) => {
  return state[NAME_SPACE].reviews;
};

export const getInitCity = createSelector(
    [getOffers],
    (offers) => {
      if (offers.length) {
        return offers[0].city;
      }
      return ``;
    }
);

export const getFilteredOffers = createSelector(
    [getCity, getOffers],
    (city, offers) => {
      return offers.filter((offer) => offer.city.name === city.name);
    }
);


export const getCities = createSelector(
    [getOffers],
    (offers) => {
      const result = [];
      for (const value of Object.values(offers)) {
        if (result.findIndex((city) => (city.name === value.city.name)) === -1) {
          result.push(value.city);
        }
        if (result.length >= CITIES_TO_SHOW) {
          return result;
        }
      }
      return result;
    }
);


