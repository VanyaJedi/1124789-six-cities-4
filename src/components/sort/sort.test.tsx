import * as React from "react";
import * as renderer from "react-test-renderer";
import Sort from "./sort";
import {sortType, sortTypeMapping} from "../../mock/test/data";

it(`Sort snapshot`, () => {
  const onChangeSortType = jest.fn();
  const opened = false;
  const onClickHandler = jest.fn();
  const tree = renderer
    .create(
        <Sort
          onChangeSortType={onChangeSortType}
          onClickHandler={onClickHandler}
          opened={opened}
          sortTypeMapping={sortTypeMapping}
          sortType={sortType}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
