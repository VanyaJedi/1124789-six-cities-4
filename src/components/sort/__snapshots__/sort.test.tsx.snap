// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort snapshot 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by
  </span>
  <span
    className="places__sorting-type"
    onClick={[MockFunction]}
    tabIndex={0}
  >
    Popular
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom "
  >
    <li
      className="places__option places__option--active"
      data-sortype="Popular"
      onClick={[Function]}
    >
      Popular
    </li>
    <li
      className="places__option "
      data-sortype="CostAsc"
      onClick={[Function]}
    >
      Price: low to high
    </li>
    <li
      className="places__option "
      data-sortype="CostDesc"
      onClick={[Function]}
    >
      Price: high to low
    </li>
    <li
      className="places__option "
      data-sortype="RateDesc"
      onClick={[Function]}
    >
      Top rated first
    </li>
  </ul>
</form>
`;
